### PR TITLE
Support for Pandas Dataframe and Series

### LINKED ISSUE #125: [FEATURE]: Return pd.DataFrame if X is DataFrame after split
**Is your feature request related to a problem? Please describe.**

If X is a DataFrame with meaningful index, X_train and X_test are ndarrays without original index.

**Use-cases/examples of this new feature**

Return X_train, X_test as pd.DataFrame, y_train, y_test as pd.Series (or DataFrame, based on what y is).

**Desired solution/workflow**

Just like this package: https://github.com/yu9824/kennard_stone

**Discussion**

**Additional context**
Thank you!

### PR DESCRIPTION
**Pull request by JacksonBurns**
This PR is attempting to add support for Pandas Dataframes and Series directly.

Resolves #125

### CODE REVIEW

**Comment by JacksonBurns** [comment]
Perhaps we could reduce the scope of this somewhat? Only support DataFrames for X and Series for y?

**Comment by kspieks** [comment]
Firstly, `panda_handla` is a 10/10 choice for variable name 👌

As a first pass, I think it's okay to assume y is a Series object for now. However, my initial thought is that we would want to extend support to Dataframe as well. It's probably also a safe first pass assumption that even if y is a dataframe, it is only 1 column for now. I believe some splitting techniques basically perform KS or SPXY or extrapolation based on the target value and this would be easiest to implement if there is only one target value per entry.

I'll be pretty busy this week, but am happy to schedule a call sometime if it's helpful to discuss details about how to implement this feature.

File: astartes/utils/array_type_helpers.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            X_val = pd.DataFrame(
+                X_val,
+                columns=dataframe_metadata["columns"],
+                index=dataframe_metadata["index"][val_idxs],
+            )
+        out.append(X_val)
+    X_test = sampler_instance.X[test_idxs]
+    if dataframe_metadata:
+        X_train = pd.DataFrame(
+            X_train,
```
Comment (kspieks) [inline astartes/utils/array_type_helpers.py:126]: I believe this should be `X_test`? Not sure if this impacts the unittests

**Review by kspieks — CHANGES_REQUESTED** [review]
Thanks for these updates! Except for one minor comment, things look great. We should be able to merge this today

File: astartes/utils/array_type_helpers.py (hunk 1 above)
Comment (kspieks) [reply astartes/utils/array_type_helpers.py:126]: Actually it looks like the unittests only test `train_X` but not `test_X`. Could we also test `test_X`?

**Pull request merged**